### DISCUSSION TITLE
Spout & Sink system for better Async Video, Sync Video, and Audio cloning

### DISCUSSION THREAD

**Discussion by Xaymar** [Xaymar/obs-stream-effects#852] [Ideas & Feedback]
The current way of having audio or video being mirrored is not exactly bug free and comes with problems, such as left over scene references and other nonsense. A better way that allowed for improved support and rendering speed would be to have a Spout/Sink system.

A Spout would be added as either an Audio or Video filter to a source, upon which everything that happened before that filter becomes available to any Audio or Video sink. Due to the caching being performed by the origin instead of the target, a lot less CPU and GPU resources are required for bigger setups, and even more massive setups can be created which take advantage of this system.

**Comment by erickskrauch** [comment]
Great and very simple idea. I would love to help with implementation, but I'm a PHP programmer 😅

I'm not sure if this would be possible (or related to this exact plugin), but ideally, it would also be great to be able to clone a video with an alternate audio track selection.

This would be useful for receiving an SRT stream where 1 track has a microphone and the second track has the game sounds to separate them, but keep in sync.
